File: kotlin/953-reverse-only-letters-Easy/Solution.kt
class Solution {
    fun reverseOnlyLetters(s: String): String {
        val itr = s.filter { it.isLetter() }.reversed().iterator()
        val res = StringBuilder()
        s.forEach {
            val c = if (it.isLetter()) itr.next() else it
            res.append(c)
        }
        return res.toString()
    }
}

File: scala/953-reverse-only-letters-Easy/Solution.scala
object Solution {
    def reverseOnlyLetters(S: String): String = {
        val reversedLetters = S.filter(_.isLetter).reverseIterator
        val res = new StringBuilder
        for (s <- S)
            if (s.isLetter) res += reversedLetters.next else res += s
        res.toString
    }
}